core/src/main/java/hudson/util/StreamTaskListener.java

```diff
@@ -56,61 +57,63 @@
  */
 public class StreamTaskListener extends AbstractTaskListener implements Serializable, Closeable {
     private PrintStream out;
-    private Charset charset;
 
     /**
-     * @deprecated as of 1.349
-     *      The caller should use {@link #StreamTaskListener(OutputStream, Charset)} to pass in
-     *      the charset and output stream separately, so that this class can handle encoding correctly,
-     *      or use {@link #fromStdout()} or {@link #fromStderr()}.
+     * Encoding used for the log content.
+     * Used before 2.1 as Jenkins was storing build log with build node's encoding.
+     * Starting with 2.1 charset conversion is done on the fly so logs are stored on master with local encoding.
```

Comment:
As mentioned earlier, I think using local encoding is a no-no. We need to use UTF-8 to make sure any characters are representable.